compiler/circle-eval-diff/src/InputDataLoader.cpp

```diff
@@ -201,18 +217,26 @@ std::unique_ptr<InputDataLoader> makeDataLoader(const std::string &file_path,
                                                 const InputFormat &format,
                                                 const std::vector<loco::Node *> &input_nodes)
 {
-  switch (format)
+  try
   {
-    case InputFormat::H5:
+    switch (format)
     {
-      return std::make_unique<HDF5Loader>(file_path, input_nodes);
+      case InputFormat::H5:
+      {
+        return std::make_unique<HDF5Loader>(file_path, input_nodes);
+      }
```

Comment:
FYI, try-catch statement should be smaller because below `DirectoryLoader` is not related to h5 things.
```suggestion
      case InputFormat::H5:
      {
        try{
          return std::make_unique<HDF5Loader>(file_path, input_nodes);
          }
          catch(const H5::Exception &e)
          {
            H5::Exception::printErrorStack();
            throw std::runtime_error("HDF5 error occurred.");
          }
      }
```